homeassistant/components/rainmachine/sensor.py

```diff
@@ -151,6 +157,8 @@ def update():
         self._dispatcher_handlers.append(
             async_dispatcher_connect(self.hass, SENSOR_UPDATE_TOPIC, update)
         )
+        await self.rainmachine.async_register_api_interest(self._api_category)
+        update()
```

Comment:
Don't do this, as this will also write the state. Better just call `await self.async_update()`